xc/xenops_server_xen.ml

```diff
@@ -2622,14 +2623,22 @@ module VBD = struct
              let open Device_common in
              let device_number = dev.frontend.devid |> Device_number.of_xenstore_key in
              let qemu_domid = this_domid ~xs in
-             let qemu_frontend = match Device_number.spec device_number with
-               | Ide, n, _ when n < 4 ->
+             let qemu_frontend =
+               let maybe_create_vbd_backend () =
                  let index = Device_number.to_disk_number device_number in
                  begin match vbd.Vbd.backend with
                    | None   -> Some (index, Empty)
                    | Some _ -> Some (index, create_vbd_frontend ~xc ~xs task qemu_domid vdi)
                  end
-               | _,_,_ -> None in
+               in
+               match Device_number.spec device_number with
+               | Ide,    n, _ when 0 <= n && n < 4 -> maybe_create_vbd_backend ()
+               | Floppy, n, _ when 0 <= n && n < 2 -> maybe_create_vbd_backend ()
+               | Ide,    n, _                      -> D.warn "qemu_frontend: Ide index should be less than 4, but got: %i" n; None
+               | Floppy, n, _                      -> D.warn "qemu_frontend: floppy index should be less than 2, but got: %i" n; None
```

Comment:
Consider changing message to something similar to "qemu_frontend: XXXX supports device numbers between 0 and Y; received %i.". This message shows both limits, uses terminology from the domain and avoid using "should" when we mean it's not supported at all.